SlevomatCodingStandard/Sniffs/TypeHints/NullabilityWhenNullDefaultValueSniff.php

```diff
@@ -0,0 +1,79 @@
+<?php declare(strict_types = 1);
+
+namespace SlevomatCodingStandard\Sniffs\TypeHints;
+
+use SlevomatCodingStandard\Helpers\TokenHelper;
+
+class NullabilityWhenNullDefaultValueSniff implements \PHP_CodeSniffer_Sniff
+{
+
+	const CODE_NULLABILITY_SYMBOL_REQUIRED = 'NullabilitySymbolRequired';
+
+	/**
+	 * @return int[]
+	 */
+	public function register(): array
+	{
+		return [
+			T_FUNCTION,
+			T_CLOSURE,
+		];
+	}
+
+	/**
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint
+	 * @param \PHP_CodeSniffer_File $phpcsFile
+	 * @param int $functionPointer
+	 */
+	public function process(\PHP_CodeSniffer_File $phpcsFile, $functionPointer)
+	{
+		$tokens = $phpcsFile->getTokens();
+		$startPointer = $tokens[$functionPointer]['parenthesis_opener'] + 1;
+		$endPointer = $tokens[$functionPointer]['parenthesis_closer'];
+
+		for ($i = $startPointer; $i < $endPointer; $i++) {
+			if ($tokens[$i]['code'] === T_VARIABLE) {
+				$parameterName = $tokens[$i]['content'];
+
+				$afterVariable = TokenHelper::findNextEffective($phpcsFile, $i + 1);
+				if ($tokens[$afterVariable]['code'] !== T_EQUAL) {
+					continue;
+				}
+
+				$afterEquals = TokenHelper::findNextEffective($phpcsFile, $afterVariable + 1);
+				if ($tokens[$afterEquals]['code'] !== T_NULL) {
+					continue;
+				}
+
+				$ignoreToken1 = array_merge(TokenHelper::$ineffectiveTokenCodes, [T_BITWISE_AND, T_ELLIPSIS]);
+				$typehint = TokenHelper::findPreviousExcluding($phpcsFile, $ignoreToken1, $i - 1, $startPointer);
+
+				if ($typehint === null || !in_array($tokens[$typehint]['code'], [T_NS_SEPARATOR, T_STRING], true)) {
+					continue;
+				};
+
+				$ignoreTokens2 = array_merge(TokenHelper::$ineffectiveTokenCodes, [T_BITWISE_AND, T_ELLIPSIS, T_NS_SEPARATOR, T_STRING]);
+				$beforeTypehint = TokenHelper::findPreviousExcluding($phpcsFile, $ignoreTokens2, $i - 1, $startPointer);
```

Comment:
The third parametr should be `$typehint - 1`. Then you don't need to ignore `T_BITWISE_AND` and `T_ELLIPSIS` here and you can use the same list of type hint tokens from line 51.